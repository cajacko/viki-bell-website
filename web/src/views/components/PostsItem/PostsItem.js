/* @flow */

import React from 'react';
import PropTypes from 'prop-types';
import momentPropTypes from 'react-moment-proptypes';
import getRouteFromSlug from 'helpers/routing/getRouteFromSlug';

class PostsItem extends React.Component {
  constructor(props: {
    title: String,
    date: {},
    image: String,
    category: String,
    imageAlt: String,
    slug: String,
  }) {
    super(props);

    this.state = { route: '' };
    this.getPostRoute = this.getPostRoute.bind(this);
  }

  state: { route: string };

  componentDidMount() {
    this.getPostRoute(this.props.slug);
  }

  componentWillReceiveProps(nextProps: {
    title: String,
    date: {},
    image: String,
    category: String,
    imageAlt: String,
    slug: String,
  }) {
    this.getPostRoute(nextProps.slug);
  }

  getPostRoute(slug: string) {
    const route = getRouteFromSlug(slug);
    this.setState({ route });
  }

  getPostRoute: () => void;

  render() {
    return (
      <article>
        <a href={this.state.route}>
          <img alt={this.props.imageAlt} src={this.props.image} />
        </a>
        <p>{this.props.date.format('YYYY-MM-DD')}</p>
        <a href={this.state.route}>
          <h2>{this.props.title}</h2>
        </a>
        <a href={this.state.route}>{this.props.category}</a>
      </article>
    );
  }
}

PostsItem.propTypes = {
  title: PropTypes.string.isRequired,
  date: momentPropTypes.momentObj.isRequired,
  image: PropTypes.string.isRequired,
  category: PropTypes.string.isRequired,
  imageAlt: PropTypes.string.isRequired,
  slug: PropTypes.string.isRequired,
};

export default PostsItem;
