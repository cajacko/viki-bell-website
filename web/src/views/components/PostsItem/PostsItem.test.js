import React from 'react';
import PostsItem from 'components/PostsItem/PostsItem';
import renderer from 'react-test-renderer';
import moment from 'moment';

it('renders correctly', () => {
  const tree = renderer.create(
    <PostsItem
      title="Viki Bell Post Title"
      date={moment()}
      image="https://unsplash.it/400/300"
      category="Life"
      imageAlt="Dummy Image"
      slug="viki-bell-post-title"
    />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});

it('renders correctly again', () => {
  const tree = renderer.create(
    <PostsItem
      title="Viki Bell Post Title 2"
      date={moment()}
      image="https://unsplash.it/200/300"
      category="Food"
      imageAlt="Ooo yeah"
      slug="viki-bell-post-title-2"
    />,
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
