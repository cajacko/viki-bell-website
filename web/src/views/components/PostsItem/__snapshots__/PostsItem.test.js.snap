// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<article>
  <a
    href="viki-bell-post-title"
  >
    <img
      alt="Dummy Image"
      src="https://unsplash.it/400/300"
    />
  </a>
  <p>
    2017-05-01
  </p>
  <a
    href="viki-bell-post-title"
  >
    <h2>
      Viki Bell Post Title
    </h2>
  </a>
  <a
    href="viki-bell-post-title"
  >
    Life
  </a>
</article>
`;

exports[`renders correctly again 1`] = `
<article>
  <a
    href="viki-bell-post-title-2"
  >
    <img
      alt="Ooo yeah"
      src="https://unsplash.it/200/300"
    />
  </a>
  <p>
    2017-05-01
  </p>
  <a
    href="viki-bell-post-title-2"
  >
    <h2>
      Viki Bell Post Title 2
    </h2>
  </a>
  <a
    href="viki-bell-post-title-2"
  >
    Food
  </a>
</article>
`;
